ivy/func_wrapper.py

```diff
@@ -556,6 +557,26 @@ def _outputs_to_ivy_arrays(*args, **kwargs):
     return _outputs_to_ivy_arrays
 
 
+def handle_backend_invalid(fn: Callable) -> Callable:
+    @functools.wraps(fn)
+    def _handle_backend_invalid(*args, **kwargs):
+        from ivy.utils.backend.handler import _determine_backend_from_args
+
+        for arg in args:
+            if ivy.current_backend() != _determine_backend_from_args(arg):
+                raise ivy.utils.exceptions.InvalidBackendException(
+                    f"The current backend is {ivy.current_backend()} while the"
+                    " ivy.Array used is based on the"
+                    f" {_determine_backend_from_args(arg)} backend."
```

Comment:
We can cache the result from `_determine_backend_from_args` instead of calling it once more.